File: src/main/scala/net/aicomp/terraforming/entity/Installation.scala
package net.aicomp.terraforming.entity

object Installation {

  val initial = new Installation("initial", 0, 0, 3)
  val robotmaker = new Installation("robotmaker", 50, 4, 3)
  val excavator = new Installation("excavator", 20, 4, 3)
  val tower = new Installation("tower", 25, 5, 3)
  val bridge = new Installation("bridge", 15, 4, 3)
  val house = new Installation("house", 10, 4, 3)
  val town = new Installation("town", 10, 9, 3)

  val buildables = List(robotmaker, excavator, tower, bridge, house, town)
}

class Installation(val name: String, val robotCost: Int, val materialCost: Int, val score: Int) {
  def head = name.head

  override def toString = name
}

File: SampleAI/Java/net/aicomp/terraforming/ai/entity/Installation.java
package net.aicomp.terraforming.ai.entity;

public enum Installation {
  Initial(0, 0, 3),
  Robotmaker(50, 4, 3),
  Excavator(20, 4, 3),
  Tower(25, 5, 3),
  Bridge(15, 4, 3),
  House(10, 4, 3),
  Town(10, 9, 3),
  None(0, 0, 0);

  public final int robotCost;
  public final int materialCost;
  public final int score;

  private Installation(int robotCost, int materialCost, int score) {
    this.robotCost = robotCost;
    this.materialCost = materialCost;
    this.score = score;
  }
}
